src/schematools/types.py

```diff
@@ -37,6 +40,86 @@
 
 ST = TypeVar("ST", bound="SchemaType")
 Json = Union[str, int, float, bool, None, Dict[str, Any], List[Any]]
+Ref = str
+
+
+@total_ordering
+class SemVer:
```

Comment:
In the amsterdam-schema repo this library is used: https://pypi.org/project/semver/ Could that have been used here as well, or was it not adequate for this use-case?

Reply:
Didn't knew that library existed. Could have used that. However it would still involve writing code. As it is, `semver.version.Version` supports additional labels for pre-release and build metadata; we specifically don't. Hence I would need to subclass it to put those restrictions in place.

In addition `SemVer` is-a `str`. Hence I can use it anywhere a `str` is expected without any casts.

So the subclassing + addional casts might amount to the same amount of code (not counting the unit tests). So I'd like to keep it in. But if you are more comfortable with the library I will change things.

Reply:
After reading the rest of the code, I now see that this `SemVer` implementation is really tailored to our use-case. Maybe we could do it the other way around now, and use our own version in the amsterdam-schema repo :-)

Reply:
I might extract it into `more-ds`, but will do so in a separate PR.